### PR TITLE
Ccd 4590 solicitors queries

### PR DESCRIPTION
**Pull request by kerry-anne-mclaughlin**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CCD-4590

### Change description ###

CCD will create a new complex basetype called CaseQueriesCollection.

This complex basetype will contain a single field called CaseMessages, which is of type [Collection](https://tools.hmcts.net/confluence/display/RCCD/CCD+Supported+Field+Types#CCDSupportedFieldTypes-ccdTypeCollection) of a seperate new complex basetype called CaseMessage.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: repository/src/main/resources/db/migration/V20230724_4590__CCD-4590_CreateCaseQueriesCollection.sql [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+values (now(), 'CaseQueriesCollection', 1,
+(select id from field_type where reference = 'Complex'
+and jurisdiction_id is null
+and version = (select max(version)
+from field_type where reference = 'Complex'
+and jurisdiction_id is null
+and base_field_type_id is null))
+);
+
+insert into complex_field (reference, label, security_classification, field_type_id, complex_field_type_id)
```
Comment (RebeccaBaker) [inline repository/src/main/resources/db/migration/V20230724_4590__CCD-4590_CreateCaseQueriesCollection.sql:11]: Not sure an Id is required here?

File: repository/src/main/resources/db/migration/V20230723_4590__CCD-4590_CreateCaseMessage.sql [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+insert into field_type (created_at, reference, version, base_field_type_id, collection_field_type_id)
+values (now(), 'documentCollection', 1,
+    (select id from field_type where reference = 'Collection'
+        and jurisdiction_id is null
+        and version = (select max(version)
+        from field_type where reference = 'Collection'
+        and jurisdiction_id is null
+        and base_field_type_id is null)),
+    (select id from field_type where reference = 'document'
```
Comment (RebeccaBaker) [inline repository/src/main/resources/db/migration/V20230723_4590__CCD-4590_CreateCaseMessage.sql:34]:
```suggestion
    (select id from field_type where reference = 'Document'
```

File: repository/src/main/resources/db/migration/V20230723_4590__CCD-4590_CreateCaseMessage.sql [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+(select id from field_type where reference = 'Text' and version = 1 and jurisdiction_id is null),
+(select id from field_type where reference = 'CaseMessage' and version = 1 and jurisdiction_id is null));
+
+insert into complex_field (reference, label, security_classification, field_type_id, complex_field_type_id)
+values ('body', 'body', 'PUBLIC',
+(select id from field_type where reference = 'Text' and version = 1 and jurisdiction_id is null),
+(select id from field_type where reference = 'CaseMessage' and version = 1 and jurisdiction_id is null));
+
+insert into field_type (created_at, reference, version, base_field_type_id, collection_field_type_id)
+values (now(), 'documentCollection', 1,
```
Comment (RebeccaBaker) [inline repository/src/main/resources/db/migration/V20230723_4590__CCD-4590_CreateCaseMessage.sql:27]:
```suggestion
values (now(), 'DocumentCollection', 1,
```

File: repository/src/main/resources/db/migration/V20230723_4590__CCD-4590_CreateCaseMessage.sql [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        and jurisdiction_id is null
+        and base_field_type_id is null)),
+    (select id from field_type where reference = 'document'
+        and jurisdiction_id is null
+        and version = 1)
+);
+
+insert into complex_field (reference, label, security_classification, field_type_id, complex_field_type_id)
+values ('attachments', 'attachments', 'PUBLIC',
+    (select id from field_type where reference = 'documentCollection' and version = 1 and jurisdiction_id is null),
```
Comment (RebeccaBaker) [inline repository/src/main/resources/db/migration/V20230723_4590__CCD-4590_CreateCaseMessage.sql:41]:
```suggestion
    (select id from field_type where reference = 'DocumentCollection' and version = 1 and jurisdiction_id is null),
```

File: repository/src/main/resources/db/migration/V20230723_4590__CCD-4590_CreateCaseMessage.sql (hunk 2 above)
Comment (RebeccaBaker) [reply repository/src/main/resources/db/migration/V20230723_4590__CCD-4590_CreateCaseMessage.sql:34]: As per https://github.com/hmcts/ccd-definition-store-api/blob/master/repository/src/main/resources/db/migration/V0001__Base_version.sql#L2563-L2564 it's Document not document

**Review by RebeccaBaker — APPROVED** [review]
LGTM

**Review by RebeccaBaker — APPROVED** [review]

File: repository/src/main/resources/db/migration/V20230723_4590__CCD-4590_CreateCaseMessage.sql [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+(select id from field_type where reference = 'CaseMessage' and version = 1 and jurisdiction_id is null));
+
+insert into complex_field (reference, label, security_classification, field_type_id, complex_field_type_id)
+values ('createdOn', 'createdOn', 'PUBLIC',
+(select id from field_type where reference = 'Date' and version = 1 and jurisdiction_id is null),
+(select id from field_type where reference = 'CaseMessage' and version = 1 and jurisdiction_id is null));
+
+insert into complex_field (reference, label, security_classification, field_type_id, complex_field_type_id)
+values ('createdBy', 'createdBy', 'PUBLIC',
+(select id from field_type where reference = 'DateTime' and version = 1 and jurisdiction_id is null),
```
Comment (danlysiak) [inline repository/src/main/resources/db/migration/V20230723_4590__CCD-4590_CreateCaseMessage.sql:66]:
```suggestion
(select id from field_type where reference = 'Text' and version = 1 and jurisdiction_id is null),
```

File: repository/src/main/resources/db/migration/V20230723_4590__CCD-4590_CreateCaseMessage.sql [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+(select id from field_type where reference = 'CaseMessage' and version = 1 and jurisdiction_id is null));
+
+insert into complex_field (reference, label, security_classification, field_type_id, complex_field_type_id)
+values ('hearingDate', 'hearingDate', 'PUBLIC',
+(select id from field_type where reference = 'Date' and version = 1 and jurisdiction_id is null),
+(select id from field_type where reference = 'CaseMessage' and version = 1 and jurisdiction_id is null));
+
+insert into complex_field (reference, label, security_classification, field_type_id, complex_field_type_id)
+values ('createdOn', 'createdOn', 'PUBLIC',
+(select id from field_type where reference = 'Date' and version = 1 and jurisdiction_id is null),
```
Comment (danlysiak) [inline repository/src/main/resources/db/migration/V20230723_4590__CCD-4590_CreateCaseMessage.sql:61]:
```suggestion
(select id from field_type where reference = 'DateTime' and version = 1 and jurisdiction_id is null),
```

File: repository/src/main/resources/db/migration/V20230724_4590__CCD-4590_CreateCaseQueriesCollection.sql [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        from field_type where reference = 'Collection'
+        and jurisdiction_id is null
+        and base_field_type_id is null)),
+    (select id from field_type where reference = 'CaseMessage'
+        and jurisdiction_id is null
+        and version = 1)
+);
+
+insert into complex_field (reference, label, security_classification, field_type_id, complex_field_type_id)
+values ('CaseMessages', 'Case messages', 'PUBLIC',
```
Comment (danlysiak) [inline repository/src/main/resources/db/migration/V20230724_4590__CCD-4590_CreateCaseQueriesCollection.sql:25]:
```suggestion
values ('caseMessages', 'Case messages', 'PUBLIC',
```
I think the ticket missed updating this to align with the defined frontend model.

**Review by danlysiak — COMMENTED** [review]
Some minor comments

**Review by RebeccaBaker — APPROVED** [review]

**Review by danlysiak — APPROVED** [review]

**Pull request merged**